crates/matrix-sdk-common/src/timeout.rs

```diff
@@ -41,17 +41,18 @@ impl Error for ElapsedError {}
 /// an error.
 pub async fn timeout<F, T>(future: F, duration: Duration) -> Result<T, ElapsedError>
 where
-    F: Future<Output = T> + Unpin,
+    F: Future<Output = T>,
 {
     #[cfg(not(target_arch = "wasm32"))]
+    #[allow(clippy::disallowed_methods)]
     return tokio_timeout(duration, future).await.map_err(|_| ElapsedError(()));
 
     #[cfg(target_arch = "wasm32")]
     {
         let timeout_future =
             TimeoutFuture::new(u32::try_from(duration.as_millis()).expect("Overlong duration"));
 
-        match select(future, timeout_future).await {
+        match select(std::pin::pin!(future), timeout_future).await {
```

Comment:
This is a nice change! I would have preferred it in a separate PR, but I'm not going to insist on it.

Reply:
I wanted everything in separate PRs, but squashed all of them into one after clippy experiments :/